allensdk/brain_observatory/behavior/behavior_ophys_session.py

```diff
@@ -56,9 +56,14 @@ def __init__(self, api=None):
         self._corrected_fluorescence_traces = None
         self._motion_correction = None
         self._segmentation_mask_image = None
+        self._eye_tracking_data = None
+
+        # Keep track of args for data processing functions that hydrate
+        # ophys behavior session properties.
+        self._eye_tracking_processing_args = None
```

Comment:
Please also update the class constructors (I think only LIMS in this case, since this isn't relevant to NWB that's already processed?) and `__init__` method for the session. I think it's confusing if these parameters can only be set after the class instance is created.

Also is there a reason why you packaged these up together rather than just using the names of the arguments in the downstream functions?

Reply:
1. I don't believe NWB files exist for behavior + ophys yet, but can definitely update the LIMS class constructor and the __init__ method.

2. Are you asking why I packaged the args as a dict instead of having individual properties with setters? If so, I wanted to avoid too much property namespace pollution. If some future property has processing that can accept a bunch of params, that would be a lot of boilerplate to add.

Reply:
There definitely are NWB for ophys + behavior. That is what the BEHAVIOR_OPHYS_WRITE_NWB_QUEUE does.
allensdk.brain_observatory.behavior.write_nwb
(These have not been publicly released yet though)

Reply:
Hi @wbwakeman , do you know if the eye tracking is stored in NWB in a raw enough format to make use of these methods, or is it already packaged? If you don't know offhand that's something we should look into @njmei

Reply:
Sorry; I don't know offhand

Reply:
@kschelonka I took a quick look yesterday and it looks like the eye tracking ellipse fits are not currently stored in the NWB file. That said, it looks like it would be a separate scope of work:

https://github.com/AllenInstitute/AllenSDK/issues/1391

Reply:
I see your point about not wanting clutter, but to me it's pretty confusing if I have to pass the whole dictionary for all the eye arguments just to change one value. One other option could be going the more OO way of scikit-learn, spark transformers, etc. which have get and set param methods. Then we could handle everything in `set_params` and not worry about using properties to control it.

Reply:
I can speak firsthand from using the caiman package that the dictionary method is very confusing for the user. In the case of caiman for example I knew the name of the arguments I wanted, but I didn't know exactly where they were stored in their nested param dictionary. Right now it seems simple (just eye tracking), but it can quickly get more complex. With this proposed method we are requiring users to not only know the keys of the arguments, but also the names of the dictionary groupings to which they belong.

Reply:
Okay, this is done. Feel free to unresolve if anything is still unsatisfactory.